product_default_image/models/abstract_product_image.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, models
+
+from ..image_constants import IMAGE_FIELDS
+
+
+class AbstractProductImage(models.AbstractModel):
+
+    _name = 'abstract.product.image'
+    _description = 'Abstract Product Image'
+
+    @api.model
+    def _vals_get_images(self, vals,
+                         false_filter=False, img_keys=IMAGE_FIELDS):
+        """ Returns images from given vals argument.
+
+        Args:
+            vals (dict): Dictionary containing the image fields.
+
+            false_filter (bool): Set True if wanting to filter out
+                any False image values.
+
+            img_keys (list, tuple, str): Contains string values of image
+                fields. If specifying multiple, use a list or tuple.
+
+        Returns:
+            list: Returns image values of the img_keys in the vals arg
+                if the keys are in the vals dict.
+
+        """
+        if not isinstance(img_keys, (list, tuple)):
+            img_keys = [img_keys]
```

Comment:
The issue here is that, say if I pass in `'image_medium'` as my img_keys argument, tuples must be at least length 2, so it will be left as just a string even if I try to convert to a tuple. The extra logic to account for that is unnecessary imo.